test/integration_test.py

```diff
@@ -348,6 +353,36 @@ def _check_terra_bdbag(self, response: bytes):
                     # Terra doesn't allow colons in this column, but they may
                     # exist in versions indexed by TDR
                     self.assertNotIn(':', row['entity:participant_id'])
+        file_ = min(self._find_bdbag_drs_uris(rows), key=itemgetter('file_size'))
+        log.info('Downloading file %r from TDR (%i bytes)',
+                 file_['file_name'], file_['file_size'])
+        response_size = self._download_tdr_drs_file(
+            drs_client=self.azul_client.repository_plugin(catalog).drs_client(),
+            drs_uri=file_['file_drs_uri']
+        )
+        self.assertEqual(response_size, file_['file_size'])
+
+    def _find_bdbag_drs_uris(self, rows) -> Iterable[Dict[str, Union[str, int]]]:
+        drs_suffix = '__file_drs_uri'
+        header, *rows = rows
+        for column in header.keys():
+            if column.endswith(drs_suffix):
+                prefix = column.partition(drs_suffix)[0]
+                for row in rows:
+                    if row[column]:
+                        yield {
+                            'file_drs_uri': row[column],
+                            'file_name': row[prefix + '__file_name'],
+                            'file_size': int(row[prefix + '__file_size'])
+                        }
+
+    @lru_cache
+    def _download_tdr_drs_file(self, drs_client: DRSClient, drs_uri: str) -> int:
+        url = drs_client.get_object(drs_uri)
+        self.assertEqual(str(furl(url).scheme), 'https')
+        file_response = drs_client.http_client.request('GET', url)
+        self.assertEqual(file_response.status, 200)
```

Comment:
by convention expected value goes first

Reply:
Understood. There are many violations of this convention throughout the tests.